src/library_dylink.js

```diff
@@ -301,19 +301,28 @@ var LibraryDylink = {
   // that originate from side modules since these are not known at JS
   // generation time.
   $createInvokeFunction__internal: true,
-  $createInvokeFunction__deps: ['$dynCall', 'setThrew'],
+  $createInvokeFunction__deps: ['$dynCall'
+#if !DISABLE_EXCEPTION_CATCHING || SUPPORT_LONGJMP == 'emscripten'
+    , 'setThrew'
+#endif
+  ],
   $createInvokeFunction: function(sig) {
     return function() {
       var sp = stackSave();
       try {
         return dynCall(sig, arguments[0], Array.prototype.slice.call(arguments, 1));
       } catch(e) {
         stackRestore(sp);
-        // Exceptions thrown from C++ exception will be integer numbers.
-        // longjmp will throw the number Infinity. Re-throw other types of
-        // exceptions using a compact and fast check.
+        // Create a try-catch guard that rethrows the Emscripten EH exception, if any.
+#if !DISABLE_EXCEPTION_CATCHING || SUPPORT_LONGJMP == 'emscripten'
+        // Exceptions thrown from C++ will be an integer number and longjmp will throw
+        // the number Infinity. Use the compact and fast "e !== e+0" test to check if 
+        // e was not a Number.
         if (e !== e+0) throw e;
         _setThrew(1, 0);
+#else
+        throw e;
```

Comment:
In which case do we want to rethrow the exception in invokes? We didn't have that option before regardless of exception/SjLj mode.

Reply:
I think the discussion in https://github.com/emscripten-core/emscripten/pull/18535#discussion_r1096510326 clarifies this a bit.